features/329-principal-apis/api.md

```diff
@@ -267,4 +292,56 @@ expectEquals(identities, [
     },
 ]);
 
+```
+
+### Replace login ID
+```typescript
+let identities: Identity[];
+let currentIdentity: Identity;
+
+identities = await listIdentities();
+expectEquals(identities, [
+    {
+        id: "1431CB1E-8A2F-4A44-874E-70C3DB3EE043",
+        type: "password",
+        loginIDKey: "username",
+        loginID: "test1",
+        realm: "default",
+        metadata: {}
+    },
+]);
+currentIdentity = (await whoami()).identity;
+expectEquals(currentIdentity, {
+    id: "1431CB1E-8A2F-4A44-874E-70C3DB3EE043",
+    type: "password",
+    loginIDKey: "username",
+    loginID: "test1",
+    realm: "default",
+    metadata: {}
+});
+
+// after re-authentication (otherwise the call would fail):
+await replaceLoginID("test1", {"username": "test2"});
```

Comment:
If we care about security, maybe we should include in this spec, that we will keep an audit trail of updating / adding / delete of login ID? It is a common practice for system with security requirements.

(Similar measure goes to username, but that would be another features from loginID, it is even more common for username to have some notes like "last changed N months ago" etc)

Reply:
Yes I agree all security critical operations, including adding/removing login ID and linking/unlinking OAuth account, should produce audit trails. However, I don't see this feature specified for version next, and the existing spec for v1 seems to be not detailed enough. Would we list all events we track when we specify audit trail in future?

Reply:
Maybe can help create a feature issue about user operation audit, and list current known operations like the above there first?

Reply:
@kiootic pls mark it as resolve when the issue mentioned is created

Reply:
ref #340